### PR TITLE
Fix kubectl checkerr race

### PR DESCRIPTION
**Pull request by tnozicka**
**What type of PR is this?**
/kind bug
/kind flake

**What this PR does / why we need it**:
`t.Fatalf` and similar are not threadsafe so we cannot set it to a global `fatalErrHandler` which is called from goroutines. Whenever this code was reached the race detector went crazy and did not write out the important message.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/cc @soltysh @kubernetes/sig-api-machinery-bugs

### CODE REVIEW

File: pkg/kubectl/cmd/apply/apply_test.go [hunk 1]
```diff
@@ -68,6 +69,12 @@ var (
 	codec = scheme.Codecs.LegacyCodec(scheme.Scheme.PrioritizedVersionsAllGroups()...)
 )
 
+func init() {
+	klog.InitFlags(nil)
+
+	cmdtesting.InitTestErrorHandler()
```
Comment (apelisse) [inline pkg/kubectl/cmd/apply/apply_test.go]: Does it have to be initialized in `init` rather than each method?

File: pkg/kubectl/cmd/apply/apply_test.go [hunk 2]
```diff
@@ -68,6 +69,12 @@ var (
 	codec = scheme.Codecs.LegacyCodec(scheme.Scheme.PrioritizedVersionsAllGroups()...)
 )
 
+func init() {
+	klog.InitFlags(nil)
```
Comment (apelisse) [inline pkg/kubectl/cmd/apply/apply_test.go]: How is that related, why is it needed now?

File: pkg/kubectl/cmd/apply/apply_test.go (hunk 1 above)
Comment (tnozicka) [reply pkg/kubectl/cmd/apply/apply_test.go]: if you initialize it in each method, given it's setting the global variable, seems like a race to me

File: pkg/kubectl/cmd/apply/apply_test.go (hunk 2 above)
Comment (tnozicka) [reply pkg/kubectl/cmd/apply/apply_test.go]: It doesn't relate, but after the switch from glog to klog every end user binary now needs to initialize it to set up the flags, the switch didn't fix it in tests. every _test.go is the binary in the end. You'd end up using it as `go test -race ./pkg/kubectl/cmd/get -v -run='TestWatchOnlyList' -timeout=3s -args -logtostderr=true -v=9` but without those flags registered you can't. (glog registered those flags when included in its init, klog doesn't)

**Comment by tnozicka** [comment]
the `t.Errorf` (unlike `t.Fatalf`) is actually fine to be called from a goroutine up to a point - if you e.g. do `go cmd.Run` in a unit test (because stuff like watches), and there is no way known to me how to voluntarily stop the `cmd.Run` because it doesn't have context nor stop channel, you end up leaving the goroutine in the background causing an error because it can call `t.Errror` through the fatalHandler after the test ended which results in a panic because it is failing a test that already completed

**Comment by tnozicka** [comment]
I am not actually sure the change is worth it. I think setting the global variable from multiple tests running in parallel is racy, but I don't recall if I've actually hit it yesterday or not, and I can't hit it now. Switching the t.Errorf to panic on its own will just convert best effort error+panic into a consistent panic. And I've went around it by actually using `RunE` in the other PR so the handler isn't hit in my case at all.

**Comment by apelisse** [comment]
I do like it. Any other changes you want to make? I'd be happy to lgtm.
